src/helpers/redirects.ts

```diff
@@ -0,0 +1,164 @@
+import { NetlifyConfig } from '@netlify/build'
+import { yellowBright } from 'chalk'
+import { readJSON } from 'fs-extra'
+import { NextConfig } from 'next'
+import { PrerenderManifest } from 'next/dist/build'
+import { outdent } from 'outdent'
+import { join } from 'pathe'
+
+import { HANDLER_FUNCTION_PATH, HIDDEN_PATHS, ODB_FUNCTION_PATH } from '../constants'
+
+import { netlifyRoutesForNextRoute } from './utils'
+
+const generateLocaleRedirects = ({
+  i18n,
+  basePath,
+  trailingSlash,
+}: Pick<NextConfig, 'i18n' | 'basePath' | 'trailingSlash'>): NetlifyConfig['redirects'] => {
+  const redirects: NetlifyConfig['redirects'] = []
+  // If the cookie is set, we need to redirect at the origin
+  redirects.push({
+    from: `${basePath}${trailingSlash ? '/' : ''}`,
+    to: HANDLER_FUNCTION_PATH,
+    status: 200,
+    force: true,
+    conditions: {
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore Incorrect type
+      Cookie: ['NEXT_LOCALE'],
+    },
+  })
+  i18n.locales.forEach((locale) => {
+    if (locale === i18n.defaultLocale) {
+      return
+    }
+    redirects.push({
+      from: `${basePath}/`,
+      to: `${basePath}/${locale}/`,
+      status: 301,
+      conditions: {
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore Incorrect type
```

Comment:
So many ignores

<img src="https://c.tenor.com/HHbDUjAOYE0AAAAC/worried-kermit.gif" alt="worried kermit gif"/>

Reply:
I'm going to fix the `@netlify/build` types

Reply:
https://github.com/netlify/build/pull/3908